### ISSUE TITLE
support setting bytes for stdin

### ISSUE THREAD

**Issue by tiziano88** [wasmerio/wasmer-go#405]
### Motivation

It is quite common to have to provide a pre-determined stdin value to a Wasm module, without having to expose a full file system to it.

### Proposed solution

leverage this function:

https://docs.wasmtime.dev/c-api/wasi_8h.html#a525abd98ade58887a969b796ea05468e

would require modifying

https://github.com/wasmerio/wasmer-go/blob/ca60a4524acf4fc518ced3bfd818d960804a0bee/wasmer/packaged/include/wasmer.h#L705-L707

and

https://github.com/wasmerio/wasmer-go/blob/ca60a4524acf4fc518ced3bfd818d960804a0bee/wasmer/wasi.go#L196-L202

### Alternatives

The best alternative at the moment is passing the value as argument on the command line, but that's not great for large and / or binary values.

**Comment by jaime-amate** [comment]
I'm currently facing the same issue on my project. I'd be quite handy having somehow a `setStdIn` function.

Could you explain on more further details about the workaround to pass input data to the wasi module execution?

**Comment by tiziano88** [comment]
For the time being I just serialize the data as a string (base64 if necessary) and pass it as the first command line argument to the Wasm module, which then reads it as `argv[1]`

**Comment by jaime-amate** [comment]
Thanks for the explanation, I think this wouldn't even work in my use case :(

I'm trying to embed a `.wasm` module compiled by [Javy](https://github.com/bytecodealliance/javy) and according to this [issue](https://github.com/bytecodealliance/javy/issues/592), it doesn't currently expose APIs to read environment variables or command line arguments.

**Comment by tiziano88** [comment]
Right, in that case fixing this issue wouldn't help you, since this is specific to WASI, which AFAICT you are not using. I think your best bet is to declare a custom function for the Wasm module to read the data into its own memory
